### ISSUE TITLE
Unloading files

### ISSUE THREAD

**Issue by BernierCR** [arose/ngl#41]
I'm not sure the best way to unload a file. I only need one mmCIF file loaded at a time, at least in the near future. What code do I run as soon as I know I'm going to need a different file? 

Also, when I accidentally loaded a second structure, WebGL crashed. I'm not sure if that is a bug or not. It may just be that my computer ran out of memory when trying to do that. I may experiment with loading multiple large structures another time and test the limits.

**Comment by arose** [comment]
> I'm not sure the best way to unload a file. I only need one mmCIF file loaded at a time, at least in the near future. What code do I run as soon as I know I'm going to need a different file?

```JavaScript
stage.loadFile( "path" ).then( function( comp ){
  stage.removeComponent( comp );  // this removes the just loaded component
}
```

> Also, when I accidentally loaded a second structure, WebGL crashed. I'm not sure if that is a bug or not. It may just be that my computer ran out of memory when trying to do that. I may experiment with loading multiple large structures another time and test the limits.

This should not happen, loading and displaying many structures is generally supported. Depending on how you load them a crash might be caused by the creation of too many WebWorkers. So, limit the number of files loaded in parallel or load them sequentially altogether. Here is how do it for the GUI using the async.js library:

```JavaScript
async.eachLimit(
    e.dataTransfer.files,
    4,
    function( file, callback ){
        stage.loadFile( file, {
            defaultRepresentation: true
        } ).then( function(){ callback(); } );
    }
);
```

I happily consider handling that in the stage object. Would you mind opening a separate issue for it, so that we don't mix topics so much. Thanks!

**Comment by arose** [comment]
@BernierCR any news on this?

**Comment by BernierCR** [comment]
Sorry, no news. I've been focused on my job hunt and other such things.
Someday I will just really feel like doing this instead of what I should be
doing. I wish I could figure out how to crowd fund myself. It's funny,
people will crowdfund the creation of videos about science, but they don't
crowdfund the actual science.

**Comment by arose** [comment]
Good luck! There is https://experiment.com/, haven't tried it though.
